### PR TITLE
Added more docs

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: contributor-docs/components/dependencies.md [hunk 1]
```diff
@@ -3,5 +3,187 @@ parent: Firebase Components
...
+
+## Deferred Dependencies
+
+Useful for optional dependencies which have a listener-style API, i.e. the dependent component registers a
+listener with the dependency and never calls it again(instead the dependency will call the registered listener).
+A good example is `Firestore`'s use of `Auth` where `Firestore` registers a token change listener to get
+notified when a new token is available. The problem is that when `Firestore` initializes, `Auth` may not be
+present in the app, and is instead part of a dynamic module that can be loaded at runtime on demand.
+
+To solve this problem, Components have a notion of a `Deferred` dependency. A deferred is defined as follows:
```
Comment (rlazo) [inline contributor-docs/components/dependencies.md:120]:
```suggestion
To solve this problem, Components have a notion of the of a `Deferred` dependency. A deferred is defined as follows:
```

File: contributor-docs/components/dependencies.md [hunk 2]
```diff
@@ -3,5 +3,187 @@ parent: Firebase Components
...
+  interface DeferredHandler<T> {
+    @DeferredApi
+    void handle(Provider<T> provider);
+  }
+
+  void whenAvailable(DeferredHandler<T> handler);
+}
+```
+
+To use it a component needs to use `Dependency.deferred(SomeType.class)`:
```
Comment (rlazo) [inline contributor-docs/components/dependencies.md:133]:
```suggestion
To use it a component needs to call `Dependency.deferred(SomeType.class)`:
```

File: contributor-docs/components/dependencies.md (hunk 1 above)
Comment (vkryachko) [reply contributor-docs/components/dependencies.md:119]: this does not look grammatically correct

Comment (rlazo) [reply contributor-docs/components/dependencies.md:119]: Yes, absolutely. Please ignore

**Review by rlazo — APPROVED** [review]

**Pull request merged**